server/src/test/java/io/crate/planner/operators/JoinTest.java

```diff
@@ -1098,4 +1098,24 @@ public void test_relationnames_order_in_nested_loop_join() throws Exception {
         assertThat(result.relationNames().get(1).toString()).isEqualTo("doc.t1");
         assertThat(result.relationNames().get(2).toString()).isEqualTo("doc.t3");
     }
+
+    @Test
+    public void test_push_constant_join_conditions_beyond_outer_joins() throws Exception {
+        var executor = SQLExecutor.builder(clusterService)
+            .build()
+            .addTable("CREATE TABLE doc.t1 (a INT)")
+            .addTable("CREATE TABLE doc.t2 (b INT)");
+
+        QueriedSelectRelation mss = e.analyze("SELECT * FROM t1 LEFT JOIN t2 ON t1.a = t2.b AND t1.a > 1");
+
+        var plannerCtx = executor.getPlannerContext();
+        plannerCtx.transactionContext().sessionSettings().setHashJoinEnabled(false);
```

Comment:
this is not needed, and I think it can be confusing, as currently outer joins are executed only via NestedLoop.